### PR TITLE
Fixup partial graph rendering

### LINKED ISSUE #9108: Partial graph rendering
## Current behaviour

Observed in the latest master and the .NET5 conversion branch:
image

This appears to happen after any dialog is closed (e.g. interactive rebase) that causes an invalidation of the graph.

v3.5 appears to be immune, so must be a regression from the latest tinkering with the graph. @mstv can you please investigate.

### PR DESCRIPTION
**Pull request by mstv**
Fixes #9108

## Proposed changes

- Fixup `RevisionGraph.GetCachedCount`: return full `cachedCount` only if loading has really finished
- Fixup condition in `DrawItem` of `RevisionGraphColumnProvider`
- Fixup theoretical race condition with `RevisionGridControl._isReadingRevisions`

## Screenshots 

see issue

## Test methodology 

- manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 7287ca7ff5688e23ef3b0a0d242a80e744996e89
- Git 2.31.1.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET 5.0.6
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by gerhardol — COMMENTED** [review]
seem ok

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 1]
```diff
@@ -1111,6 +1111,7 @@ void OnRevisionReadCompleted()
                     {
                         await this.SwitchToMainThreadAsync();
 
+                        _gridView._revisionGraph.LoadingFinished = true;
```
Comment (RussKie) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:1114]: This looks like we're breaking encapsulation...

Comment (mstv) [reply GitUI/UserControls/RevisionGrid/RevisionGridControl.cs]: https://github.com/gitextensions/gitextensions/commit/cd52038252b67de783cbafe6fda5d5ebd57dbb8b
Or should `RevisionGridControl.OnRevisionReadCompleted` call `_gridView._revisionGraph.LoadingCompleted()` directly?

**Review by RussKie — APPROVED** [review]

**Comment by mstv** [comment]
Finally, the tests pass on AppVeyor, too.

**Comment by mstv** [comment]
@msftbot merge in 1 day

**Pull request merged**